### PR TITLE
chore(atomic): add UT for suggestions-common.ts

### PR DESCRIPTION
**Pull request by vhilotii-coveo**
This PR adds KIT-4999 comprehensive unit tests for the `suggestions-common.ts` utility file in the Atomic package. The tests ensure that all exported functions are thoroughly validated for correctness and edge cases.

**Tested Functions**:
- `#elementHasNoQuery`
- `#elementHasQuery`
- `#dispatchSearchBoxSuggestionsEvent`

**Key Changes**:
- Added suggestions-common.spec.ts to test the utility functions.
- Mocked dependencies like `closest` and event utilities to isolate the tests.
- Verified edge cases such as undefined, empty strings, and invalid parent elements.

### CODE REVIEW

File: packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  type SearchBoxSuggestionElement,
+  type SearchBoxSuggestionsEvent,
+} from './suggestions-common';
+
+vi.mock('../../../utils/event-utils', {spy: true});
+vi.mock('../../../utils/dom-utils', {spy: true});
+
+describe('suggestions-common', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
```
Comment (vhilotii-coveo) [inline packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts:16]: To be removed

File: packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: 'search query',
+      };
+
+      expect(elementHasQuery(element)).toBe(true);
+    });
+
+    it('should return true when element has whitespace-only query', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts:106]: Interesting... I'm not sure this is the desired behavior 🤔 

Could you check how the method is used in the project and see if this makes sense, or if we should instead treat whitespace only strings as no query?

Similar comment for elementHasNoQuery

File: packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      const element: SearchBoxSuggestionElement = {
+        key: 'test-key',
+        content: document.createElement('div'),
+        query: '   ',
+      };
+
+      expect(elementHasQuery(element)).toBe(true);
+    });
+
+    it('should return true when element has numeric zero as query', () => {
```
Comment (fbeaudoincoveo) [inline packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts:116]: We don't have a test equivalent to this one for the elementHasNoQuery method... but at this rate we'd also want to check the 'null' and 'undefined' strings 😅 

TBH I think this test is overkill and we should just remove it.

**Review by fbeaudoincoveo — APPROVED** [review]
Nice!

Looking at suggestions-common.ts it seems like a kind of "carry-all" module... it exports interfaces / types and util methods.

Maybe it would be worth doing a bit of refactoring to split this into more aptly named modules.

If you agree, please create another Jira to do this!

File: packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts (hunk 2 above)
Comment (vhilotii-coveo) [reply packages/atomic/src/components/common/suggestions/suggestions-common.spec.ts:106]: that's insightful, will double check, thank you!
